cmd/kpod/ps.go

```diff
@@ -0,0 +1,350 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+	"text/tabwriter"
+	"time"
+
+	"github.com/docker/go-units"
+	specs "github.com/opencontainers/runtime-spec/specs-go"
+
+	"k8s.io/apimachinery/pkg/fields"
+
+	"github.com/kubernetes-incubator/cri-o/cmd/kpod/formats"
+	"github.com/kubernetes-incubator/cri-o/libkpod"
+	"github.com/pkg/errors"
+	"github.com/urfave/cli"
+)
+
+type psOptions struct {
+	all     bool
+	filter  string
+	format  string
+	last    int
+	latest  bool
+	noTrunc bool
+	quiet   bool
+	size    bool
+	label   string
+}
+
+type psOutputParams struct {
+	ID         string        `json:"id"`
+	Image      string        `json:"image"`
+	Command    string        `json:"command"`
+	CreatedAt  time.Time     `json:"created"`
+	RunningFor time.Duration `json:"running"`
+	Status     string        `json:"status"`
+	Ports      string        `json:"ports"`
+	Size       string        `json:"size"`
+	Names      string        `json:"names"`
+	Labels     string        `json:"labels"`
+	Label      string        `json:"specificLabel"`
+	Mounts     string        `json:"mounts"`
+	exitCode   int32
+}
+
+var (
+	psFlags = []cli.Flag{
+		cli.BoolFlag{
+			Name:  "all, a",
+			Usage: "Show all the containers, default is only running containers",
+		},
+		cli.StringFlag{
+			Name:  "filter, f",
+			Usage: "Filter output based on conditions given",
+		},
+		cli.StringFlag{
+			Name:  "format",
+			Usage: "Pretty-print containers using a Go template",
+		},
+		cli.IntFlag{
+			Name:  "last, n",
+			Usage: "Display the n last created containers (all states)",
+		},
+		cli.BoolFlag{
+			Name:  "latest, l",
+			Usage: "Show the latest container created (all states)",
+		},
+		cli.BoolFlag{
+			Name:  "no-trunc",
+			Usage: "Displays the extended informations",
+		},
+		cli.BoolFlag{
+			Name:  "quiet, q",
+			Usage: "Display the numeric IDs only",
+		},
+		cli.BoolFlag{
+			Name:  "size, s",
+			Usage: "Display the total file sizes",
+		},
+	}
+	psDescription = "Prints out information about the containers"
+	psCommand     = cli.Command{
+		Name:        "ps",
+		Usage:       "List containers",
+		Description: psDescription,
+		Flags:       psFlags,
+		Action:      psCmd,
+		ArgsUsage:   "",
+	}
+)
+
+type outStruct struct {
+	output []psOutputParams
+	opts   psOptions
+}
+
+func (sOut outStruct) Out() error {
+	var w *tabwriter.Writer
+	if len(sOut.output) > 0 {
+		w = tabwriter.NewWriter(os.Stdout, 0, 0, 1, ' ', 0)
+		defer w.Flush()
+
+		if !sOut.opts.quiet {
+			if sOut.opts.size {
+				fmt.Fprintln(w, "CONTAINER ID\t\tIMAGE\t\tCOMMAND\t\tCREATED\t\tSTATUS\t\tPORTS\t\tNAMES\t\tSIZE\t")
```

Comment:
Can simplify this to only one call to Fprintln if you append SIZE to the string if the bool is set, and then output from it - like this:

`var str = "..."; if sOut.opts.size { str += "SIZE\t" } fmt.Fprintln(w, str)`